test/test-utf8.cpp

```diff
@@ -0,0 +1,99 @@
+#include <gtest/gtest.h>
+
+#include "utf8.h"
+
+TEST(Utf8Test, Ascii)
+{
+    std::string str = "a";
+    Url::Utf8::codepoint_t value = 97;
+    EXPECT_EQ(value, Url::Utf8::toCodepoint(str));
+    EXPECT_EQ(str, Url::Utf8::fromCodepoint(value));
+}
+
+TEST(Utf8Test, Umlaut)
+{
+    std::string str = "ü";
+    Url::Utf8::codepoint_t value = 252;
+    EXPECT_EQ(value, Url::Utf8::toCodepoint(str));
+    EXPECT_EQ(str, Url::Utf8::fromCodepoint(value));
+}
+
+TEST(Utf8Test, IntegralSymbol)
+{
+    std::string str = "∫";
+    Url::Utf8::codepoint_t value = 8747;
+    EXPECT_EQ(value, Url::Utf8::toCodepoint(str));
+    EXPECT_EQ(str, Url::Utf8::fromCodepoint(value));
+}
+
+TEST(Utf8Test, PhaistosDiscSignEagle)
+{
+    std::string str = "𐇮";
```

Comment:
I prefer source code to be ASCII, and thus use the escaped form rather than a literal character. Same for the test strings above.